src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessaryParenthesesCheck.java

```diff
@@ -88,6 +88,10 @@
  * }
  * if ((++f) &gt; g &amp;&amp; a) { // violation, unnecessary paren
  * }
+ * if ((a &amp; d) | b) {} // violation, unnecessary paren
```

Comment:
More curious, we are saying we have only partial support. Is there something missing for support for the tokens we are adding?

@romani Do we have something somewhere that says what full support is?

Reply:
Better to extend doc to define what we can not do

Reply:
can i change line no 70 as 
`The full support for operator precedence includes cases of the following type:`
and will remove the new line added and create a new section with text 
`The partial support for operator precedence includes cases of the following type:`
and add the example of new token

Reply:
look like
![image](https://user-images.githubusercontent.com/97679350/232098793-0a6caef2-b430-4eca-b408-29349e637957.png)

Reply:
I think it should say what the partial is not supporting. How am I to know as a user if my case is something not supported or is supported and just broken? One case I shouldn't report an issue since it isn't supported and the other I should report it since it is broken.

Reply:
please give look

Reply:
I am still confused by the new text. It is not clear to me what you are saying is not supported under `Partial not supported for operator`. Are you showing false positives and false negatives? What is exactly not supported by this PR with the new tokens?
If there are things supported by this PR, an example or 2 should be added to `The partial support for operator precedence includes `.

I would expect things like: `violation not reported because we are not type aware` and `violation reported but this could fail if X was Y`.

Reply:
@rnveach  This pr should support for three new token but they are not going to be supported by default user have to add this token by themself.   

> I would expect things like: violation not reported because we are not type aware

it is already mentioned https://checkstyle.org/config_coding.html#UnnecessaryParentheses:~:text=The%20check%20is%20not%20%22type%20aware%22%2C%20that%20is%20to%20say%2C%20it%20can%27t%20tell%20if%20parentheses%20are%20unnecessary%20based%20on%20the%20types%20in%20an%20expression.%20The%20check%20is%20partially%20aware%20about%20operator%20precedence%20but%20unaware%20about%20operator%20associativity.%20It%20won%27t%20catch%20cases%20such%20as%3A 

violation reported but this could fail if X was Y.

I have added separate section for it and put the example to show what is violated 

I will change the new text some thing like  `The  partial support for operator precedence includes bitwise operator but they are not supported by default`
is it fine

Reply:
I am looking for a response at https://github.com/checkstyle/checkstyle/pull/12703#discussion_r1166947780 then